xds/src/main/java/io/grpc/xds/Bootstrapper.java

```diff
@@ -111,7 +112,11 @@ static BootstrapInfo parseConfig(String rawData) throws IOException {
           channelCredsOptions.add(creds);
         }
       }
-      servers.add(new ServerInfo(serverUri, channelCredsOptions));
+      List<?> serverFeatures = JsonUtil.getList(serverConfig, "server_features");
+      if (serverFeatures != null && serverFeatures.contains(XDS_V3_SERVER_FEATURE)) {
```

Comment:
Given that the format of server_features is undefined. Maybe it's better to just parse it to a String list.

The parser should not try to interpret what is being parsed. It's just a list of string. "xds_v3" should be defined in the consumer class, which is `XdsChannelFactory` (e.g., `String PROTOCOL_VERSION = "xds_v3`). The channel factory looks up the parsed server features and check if the expected protocol version setting is there.

Reply:
The log here should just print out what is parsed literally.

Reply:
> Given that the format of server_features is undefined. Maybe it's better to just parse it to a String list.

What do you mean actually? The following?
```java
List<String> serverFeatures = JsonUtil.getListOfStrings(serverConfig, "server_features");
```

Reply:
Yeah, just use `JsonUtil.getListOfStrings`.

Reply:
Done.